### PR TITLE
popup exit metamorpehus

### PR DESCRIPTION
**Pull request by bintiibrahim**
#1412 user can disable pop up in gui settings file

(ignore my last pull request - i closed it. reverted some commits)

### CODE REVIEW

File: GUI/MainWindow.xaml.cs [hunk 1]
```diff
@@ -1409,5 +1412,18 @@ private void AddContaminantXML_Click(object sender, RoutedEventArgs e)
...
             dataGridProteinDatabases.Items.Refresh();
         }
+
+        // handle window closing
+        private void MainWindow_Closing(object sender, CancelEventArgs e)
+        {
+            if (!GuiGlobalParams.DisableCloseWindow)
+            {
+                e.Cancel = true;
+                var exit = MessageBox.Show("Are you sure?", "Exit MetaMorpheus", MessageBoxButton.YesNo, MessageBoxImage.Question);
```
Comment (rmillikin) [inline GUI/MainWindow.xaml.cs]: change message to "Are you sure you want to exit MetaMorpheus?"

File: GUI/MainWindow.xaml.cs [hunk 2]
```diff
@@ -1409,5 +1412,18 @@ private void AddContaminantXML_Click(object sender, RoutedEventArgs e)
...
+
+                if (exit == MessageBoxResult.Yes)
+                    e.Cancel = false;
```
Comment (rmillikin) [inline GUI/MainWindow.xaml.cs]: put brackets around ifs, elses, etc, even if it's only one line. just a code style thing

File: GUI/MainWindow.xaml.cs [hunk 3]
```diff
@@ -1409,5 +1412,18 @@ private void AddContaminantXML_Click(object sender, RoutedEventArgs e)
             }
             dataGridProteinDatabases.Items.Refresh();
         }
+
+        // handle window closing
+        private void MainWindow_Closing(object sender, CancelEventArgs e)
```
Comment (rmillikin) [inline GUI/MainWindow.xaml.cs]: would be nice to have a "never ask me again" button. It sounds like that means making a custom window but that's ok if it's needed

Comment (bintiibrahim) [reply GUI/MainWindow.xaml.cs]: yeah i tried to add that but i'd have to create a custom message button. should i still do it?

Comment (bintiibrahim) [reply GUI/MainWindow.xaml.cs]: it would involve creating a new CustomMessageBox class - so idk if that's too much code

Comment (rmillikin) [reply GUI/MainWindow.xaml.cs]: yeah just make that custom class. might be nice to have around for other purposes too

File: GUI/MainWindow.xaml.cs [hunk 4]
```diff
@@ -1409,5 +1412,26 @@ private void AddContaminantXML_Click(object sender, RoutedEventArgs e)
...
+        {
+            if (!GuiGlobalParams.DisableCloseWindow)
```
Comment (rmillikin) [inline GUI/MainWindow.xaml.cs]: `if (!GuiGlobalParams.DisableCloseWindow && !GlobalVariables.Contains("DEBUG"))`

just makes it less annoying for us if you're working on a debug version and your preferences can't be saved

File: GUI/CustomMsgBox.cs [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+namespace MetaMorpheusGUI
+{
+    public partial class CustomMsgBox : Form
+    {
+        public CustomMsgBox()
+        {
+            InitializeComponent();
+            this.button1.Click += new EventHandler(Button1_Click);
+            this.button2.Click += new EventHandler(Button2_Click);
+            this.button3.Click += new EventHandler(Button3_Click);
```
[the remark this answers is not in the document]
Comment (rmillikin) [reply GUI/CustomMsgBox.cs]: nevermind, you wanted this to be a generic window, ignore my change request :)

**Pull request merged**